### DISCUSSION TITLE
Improved image management for avoiding duplicates

### DISCUSSION THREAD

**Discussion by linuxatico** [PrestaShop/PrestaShop#40151] [Ideas]
**Is your feature request related to a problem? Please describe.**
I've developed a module for batch import of images for my store products: it scans for image files in a well known directory and cURL them via the Prestashop Webservice API `/api/images/products/` .
Every image has a pattern inside its name that is a substring of one or more product references, that means that more than one product could have the same exactly image.
The result is that the same identical file is saved on the file system N times, and the file system used grows drastically.

**Describe the solution you'd like**
It would be great if Prestashop could identify such image clones (or more generally file clones) and:

1. To duplicate the file only if the file referenced by one product is altered. 
2. To remove the file only if no other products are related

**Describe alternatives you've considered**
None 

**Additional context**
None

**Comment by khouloudbelguith** [comment]
Hi @linuxatico,

Thanks for your suggestion.
@eternoendless, @PrestaShop/prestashop-core-developers what do you think of his suggestion.
Thanks!

**Comment by colinegin** [comment]
Hello @linuxatico ,

Thanks for your idea ! You're right, it's an issue we have, but we need to discuss more in depth about the solution with the core team. We will come back to you as soon as possible.

Coline

**Comment by mickaelandrieu** [comment]
Hello @linuxatico,

I think it's a good idea too, do you have worked on a solution from your side? Do you want to share your code?

Thanks

**Comment by linuxatico** [comment]
Hello,
too bad I've completely forgotten to reply this thread long ago, sorry for that.
Anyway, no: I've not tried to implement it by myself, is this ever going to make it to the next step?

**Comment by hibatallahAouadni** [comment]
Hello @linuxatico 

Sorry for the late response :sweat_smile: 
This issue is already in our debug roadmap, so could you please create a PR to fix the issue.

Thanks in advance.

**Comment by nicohery** [comment]
@hibatallahAouadni The original question has been deflected. @linuxatico wanted to share images between references without having to re upload and thus duplicate them...

**Comment by matks** [comment]
Hello,

> Anyway, no: I've not tried to implement it by myself, is this ever going to make it to the next step?

It is unlikely we'll implement a solution for this feature idea. It can be useful, but we have a roadmap and already a huge pile of work planned 😅  and this topic is off the roadmap.

I think if you want PrestaShop to implement this behavior, the best if to invest and submit a Pull Request for it 😉
